openedx/core/djangoapps/enrollments/views.py

```diff
@@ -987,3 +994,164 @@ def get_queryset(self):
+    permission_classes = (ApiKeyHeaderPermissionIsAuthenticated, )
+
+    def get(self, request):
+        """
+        Get submission history details.
+
+        **Usecases**:
+
+            Regular users can only retrieve their own submission history and users with GlobalStaff status
+            can retrieve everyone's submission history.
+
+        **Example Requests**:
+
+            GET /api/enrollment/v1/submission_history?course_id=course_id
+            GET /api/enrollment/v1/submission_history?course_id=course_id&user=username
+            GET /api/enrollment/v1/submission_history?course_id=course_id&all_users=true
+
+        **Query Parameters for GET**
+
+            * course_id: Course id to retrieve submission history.
+            * username: Single username for which this view will retrieve the submission history details.
+                If no username specified the requester's username will be used.
+            * all_users: If true and if the requester has the correct permissions,
+                retrieve history submission from every user in a course id.
+
+        **Response Values**:
+
+            If there's an error while getting the submission history an empty response will
+            be returned.
+            The submission history response has the following attributes:
+
+                * Results: A list of submission history:
+                    * course_id: Course id
+                    * course_name: Course name
+                    * user: Username
+                    * problems: List of problems
+                        * location: problem location
+                        * name: problem's display name
+                        * submission_history: List of submission history
+                            * state: State of submission.
+                            * grade: Grade.
+                            * max_grade: Maximum possible grade.
+                        * data: problem's data.
+        """
+        username = request.GET.get('username', request.user.username)
+        data = []
+        if GlobalStaff().has_user(request.user):
+            all_users = bool(request.GET.get('all', False))
+        else:
+            all_users = False
+        course_id = request.GET.get('course_id')
+
+        if not (all_users or username == request.user.username or GlobalStaff().has_user(request.user) or
+                self.has_api_key_permissions(request)):
+            return Response(data)
+
+        course_enrollments = CourseEnrollment.objects.select_related('user').filter(is_active=True)
+        if course_id:
+            if not course_id.startswith("course-v1:"):
+                course_id = "course-v1:{}".format(course_id)
+            try:
+                course_enrollments = course_enrollments.filter(
+                    course_id=CourseLocator.from_string(course_id.replace(' ', '+'))
+                ).order_by('created')
+            except KeyError:
+                return Response(data)
```

Comment:
Instead of returning empty data, we should return an error:
```suggestion
                return Response(
                    status=status.HTTP_400_BAD_REQUEST,
                    data={
                        "message": f"No course '{course_id}' found for enrollment"
                    }
                )
```